src/iocore/net/BIO_fastopen.cc

```diff
@@ -21,12 +21,72 @@
  *  limitations under the License.
  */
 
+#include <openssl/crypto.h>
+#include <openssl/opensslv.h>
+
 #include "P_Net.h"
 #include "iocore/eventsystem/SocketManager.h"
 #include "tscore/ink_assert.h"
 
 #include "BIO_fastopen.h"
 
+namespace
+{
+
+#if !defined(OPENSSL_IS_BORINGSSL)
+
+class ExData
+{
+public:
+  static int
+  idx()
+  {
+    static int idx_ = []() -> int {
+      int i = CRYPTO_get_ex_new_index(CRYPTO_EX_INDEX_BIO, 0, nullptr, _new, _dup, _free);
```

Comment:
I don't understand the necessity of `ExData` class in the first place, but even if it's needed, I don't see a reason to use `CRYPTO_get_ex_new_index` instead of `BIO_get_ex_new_index`.

Reply:
OK I will switch to BIO_get_ex_new_index, apparently it's not deprecated.  The ExData class is for scoping and to handle C++ order of initialization fiasco.